### ISSUE TITLE
Any chance for Android TV?

### ISSUE THREAD

**Issue by vcvrackideas** [Pittvandewitt/Wavelet#31]
Hi Pitt and thanks for the app! This is my favorite for my phone indeed. I'd also like to eq my speakers that connected to android tv box (mi box s). I even tried to extract apk from google play and sideload it on the box because I know it's not supported. But I have no luck even with an usb mouse connected - it doesn't catch any bands to adjust. Same with remote control both from the phone or mi box one. I may had any chance with importing eq settings but got a message that this function doesn't support on this device. 
So could you say would it be any luck and your curiosity to make it usable on Android TV devices? Thanks again for your amazing app. Cheers!

**Comment by Pittvandewitt** [comment]
Thank you for the kind words!
Currently I don't have the equipment to develop for Android TV, but I will in the future. Once that happens, I will start developing for Android TV.

**Comment by lmore377** [comment]
I know this is an old issue, but it seems to be working better now. If I enable legacy mode, I can change settings with a mouse and it actually does work. It seems like all that it needs is some better navigation support for the arrow keys (with some fiddling around I can get into menus and turn on switches and even change one band in the equalizer). I'm using a Jetstream agt418 tv box from Walmart.

**Comment by el-ef** [comment]
I was fiddeling around with a few Android TV systems in the past weeks and found that Wavelet does work actually quite well on both a Chromecast w/ Google TV as much as a Nvidia Shield Pro 2019.
The elephant in the room is the navigation, though. You won't get the full feature set unless you connect a mouse or touch pad. Switching things on and off works decently after jumping back and forth with the selection buttons of the remote. But that's all for now.

I would really appreciate an integration (GUI optimization) of Android TV since this platform has so much potential for home theatre DSP uses. For now I'm using Wavelet to eq both my speakers and headphones without the need of additional hardware in the sound path. The performance is pretty good so far. Highly recommended!

Regards

**Comment by ianken** [comment]
I'd love this. I watch video or listen to music at night using a shield pro + headphones via a fiio BTR5. Being able to apply a profile for my headphones would be cool.

**Comment by el-ef** [comment]
You can already do that. You just have to work around a little with the navigation of the remote. Or just install a remote app on your smartphone and naviagte via touch.

Technically Wavelet works just fine on my Shield Pro. It's just the navigation that is not that convenient.

**Comment by fourmealyour** [comment]
Thanks for your awesome app! I tried to install the apk on my nvidia shield and I am not able to 1) see the app on home screen, 2) not able to get it to work which I suspect it may be because the settings are blank, therefore I am unable to choose legacy mode.
Can you please provided a link or direct instructions on how to troubleshoot using the app on Nvidia Shield Pro.

kind regards

**Comment by el-ef** [comment]
Hi.
1) You need to download an alternative app drawer (sideload launcher) that will display non-stock apps. But you can also make your own custom shortcuts for the stock launcher of Shield TV without the need of additional apps: https://atvlauncher.trekgonewild.de/
2) You have to use a touch screen or mouse the navigate and activate the app. Legacy mode works for me. 

By the way: Android TV is not officially supported. And I guess that won't change too soon.

**Comment by BobbyWibowo** [comment]
2y later can confirm Wavelet still works on Android TV (Chromecast with Google TV 4K).

Navigation works fine on my end with just the Chromecast remote. D-pad movements translate well to all of the UI elements, even managed to download my headphone's AutoEq config and all that using the remote. Shortcut to the Wavelet app itself had to be via Settings > Apps because I didn't want to install third-party launcher.

Anyways, it seems Wavelet can only work on Legacy mode for Android TV yeah? I managed to grant it the DUMP permission via "ADB TV" app, and could confirm Wavelet did detect that it had the permission, but it was still unable to detect playbacks from either Jellyfin or SmartTube. But yeah, it seemingly worked fine on Legacy mode.

**Comment by MSKSuhail** [comment]
Been using it in TV (Mi box3). Navigation is rough. Need to use keyboard and mouse to configure and it is recognised as system app. So you need to go in settings to open it.

Not all versions work. I was using old version 21.09 in Android 9. All features work. Need to turn on legacy mode.

![IMG_20231223_092644046](https://github.com/Pittvandewitt/Wavelet/assets/39289619/fe4e0d62-ffc5-45a6-b8f9-0a668140421e)


![IMG_20231223_093303975](https://github.com/Pittvandewitt/Wavelet/assets/39289619/51f055c9-a6b4-4a0a-87ea-fafbcccb7658)

**Comment by erat0c** [comment]
> Thank you for the kind words! Currently I don't have the equipment to develop for Android TV, but I will in the future. Once that happens, I will start developing for Android TV.

Android TV wheeen?

**Comment by TheAymen** [comment]
Yes, the base version seems to work well with Chromecast with Google TV. 

Although I also support the idea of a dedicated Android TV version to facilitate the delivery of updates from the Google Play Store.

Thanks a lot for the great work.

**Comment by TheAymen** [comment]
After sustained use through sideloading the Chromecast with Google TV (HD), I have started to notice that there is often an issue of increased latency. I tried testing out different buffer sizes but the issue persists.

The issue is characterised by an initial stutter during playback followed by increased latency (far beyond what is expected from the buffer size setting). I use the app in legacy mode.

I understand that such issues are to be expected, considering there is no official support for Android TV, but I thought I should document the issue nonetheless in case someone has a similar issue or solution or if an Android TV is to be developed from the base version. 

Thank you.
